com.stansassets.build/Editor/Steps/IBuildStep.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace StansAssets.Build.Editor
+{    
+    /// <summary>
+    /// Single functional field of build process
+    /// </summary>
+    public interface IBuildStep
+    {    
+        /// <summary>
+        /// Run step execution
+        /// </summary>
+        /// <param name="buildContext">Data class with necessary parameters for build execution</param>
+        /// <param name="onComplete">Delegate which invokes when  step executing will be completed</param>
+        void Execute(BuildContext buildContext,Action<BuildStepResultArgs> onComplete = null);
```

Comment:
>
```suggestion
        void Execute(BuildContext buildContext, Action<BuildStepResultArgs> onComplete = null);
```

Reply:
We've talked about that, please kill `BuildStepResultArgs ` or explain why do you think it makes sense to have that.

Reply:
BuildStepResultArgs is data class and it contains all information about step execution. 
such as :

public bool IsSuccess;
public string ResultMessage;

BuildExecutor class wants know this information about Ibuild step when it is finished.In BuildExecutor we decide to continue moving to next step or finish loop.
If IBuildStep finished with error (we could check it by bool IsSuccess inside BuildStepResultArgs ), BuildExecutor displays error message(ResultMessage;) to log or other debug system.

As a solution , we can move BuildStepResultArgs to interface IBuildStep as a method (GetBuildResultArgs).
or keep it on current place.